routers/api/v1/repo/milestone.go

```diff
@@ -249,17 +239,49 @@ func DeleteMilestone(ctx *context.APIContext) {
 	//   required: true
 	// - name: id
 	//   in: path
-	//   description: id of the milestone to delete
-	//   type: integer
-	//   format: int64
+	//   description: the milestone to delete, identified by ID and if not available by name
+	//   type: string
 	//   required: true
 	// responses:
 	//   "204":
 	//     "$ref": "#/responses/empty"
 
-	if err := models.DeleteMilestoneByRepoID(ctx.Repo.Repository.ID, ctx.ParamsInt64(":id")); err != nil {
+	m := getMile(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	if err := models.DeleteMilestoneByRepoID(ctx.Repo.Repository.ID, m.ID); err != nil {
 		ctx.Error(http.StatusInternalServerError, "DeleteMilestoneByRepoID", err)
 		return
 	}
 	ctx.Status(http.StatusNoContent)
 }
+
+// getMile get milestone by ID and if not available by name
+func getMile(ctx *context.APIContext) *models.Milestone {
```

Comment:
I think you should call this getMilestoneByIDOrName and drop the reliance on ctx.